packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -44,10 +234,27 @@ export interface HttpGatewayRouteSpecOptions {
    * The VirtualService this GatewayRoute directs traffic to
    */
   readonly routeTarget: IVirtualService;
+
+  /**
+   * The default target host name to write to.
+   * Boolean is used to select either enable or disable.
+   * When enabled, rewrites the original request received at the Virtual Gateway to the destination Virtual Service hostname.
+   * When disabled, does not rewrite to destination Virtual Service hostname.
+   *
+   * @default true
+   */
+  readonly defaultHostnameRewrite?: boolean;
+
+  /**
+   * The path to rewrite.
+   *
+   * @default - rewritten to the target Virtual Service's hostname and the matched prefix is rewritten to '/'.
+   */
+  readonly pathRewrite?: HttpGatewayRouteRewrite;
```

Comment:
**Pending to respond.**

Reply:
I think I agree this is good implementation design. This would also address [this comment](https://github.com/aws/aws-cdk/pull/15395#discussion_r663231259) about the compile-time error design.

> It seems like this property only makes sense if match was provided, right?

Correct. Referring the CloudFormation doc, `Rewrite` is the property of the `Action` and `Action` represents an action to take if a match is determined

Reply:
@skinny85 - For now, I just moved `rewrite` properties to corresponding `...Match` classes.

However, I am thinking whether it is more appropriate to make `rewrite` as an optional positional arguments for specific properties in `match`.

For example:
```ts
public static exact(path: string, pathRewrite?: HttpGatewayRoutePathRewrite): HttpGatewayRoutePathMatch {
  return new HttpGatewayRoutePathSpecificMatch({ exact: path }, pathRewrite);
}

public static regex(regex: string, pathRewrite?: HttpGatewayRoutePathRewrite): HttpGatewayRoutePathMatch {
  return new HttpGatewayRoutePathSpecificMatch({ regex: regex }, pathRewrite);
}

public static prefix(prefixMatch: string, prefixRewrite?: HttpGatewayRoutePrefixRewrite): HttpGatewayRoutePathMatch {
  return new HttpGatewayRoutePathPrefixMatch(prefixMatch, prefixRewrite);
}
```

**NOTE:** Introduce additional class to `Rewrite` class to add compile-time error just like how [`MutualTlsValidation`](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-appmesh/lib/tls-validation.ts#L37) was implemented.

Reply:
This is implemented.